easybuild/framework/easyconfig/tweak.py

```diff
@@ -389,12 +389,12 @@ def pick_version(req_ver, avail_vers):
         if len(avail_vers) == 1:
             selected_ver = avail_vers[0]
         else:
-            retained_vers = [v for v in avail_vers if v <= LooseVersion(ver)]
+            retained_vers = [v for v in avail_vers if LooseVersion(v) <= LooseVersion(ver)]
             if retained_vers:
                 selected_ver = retained_vers[-1]
             else:
                 # if no versions are available that are less recent, take the least recent version
-                selected_ver = sorted([LooseVersion(v) for v in avail_vers])[0]
+                selected_ver = sorted(avail_vers, key=LooseVersion)[0]
```

Comment:
This sorting isn't actually necessary with the updated unique function

Reply:
@boegel, why was the original list the LooseVersion return values, this is causing a failing test now but I don't see a reason not to simply retain the version string

Reply:
That was most likely a bug, not a feature, since the other code paths in `pick_version` to return a string rather than a `LooseVersion` instance...